sdk/core/Azure.Core.Experimental/tests/LowLevelClient/PetStoreClient.cs

```diff
@@ -49,6 +51,22 @@ public PetStoreClient(Uri endpoint, TokenCredential credential, PetStoreClientOp
             Pipeline = HttpPipelineBuilder.Build(options, new HttpPipelinePolicy[] { new LowLevelCallbackPolicy() }, new HttpPipelinePolicy[] { authPolicy }, new ResponseClassifier());
             this.endpoint = endpoint;
             apiVersion = options.Version;
+
+            _responseClassifierCache = CreateResponseClassifierCache();
```

Comment:
My suggested approach to caching per-method response classifiers, but open to other options.

Reply:
we can use lazy properties instead of a dictionary. Every set of status codes are doing to get their own property.